doc/source/operations/control-plane-operation.rst

```diff
@@ -0,0 +1,393 @@
+.. code-block:: console
+
+   [stack@controller0 ~]$ docker exec -i mariadb mysql -u root -p -e "SHOW STATUS LIKE 'wsrep_local_state_comment'"
+   Variable_name   Value
+   wsrep_local_state_comment       Synced
+
+The password can be found using:
+
+.. code-block:: console
+
+   kayobe# ansible-vault view $KAYOBE_CONFIG_PATH/kolla/passwords.yml \
+           --vault-password-file <Vault password file path> | grep ^database
+
+Checking RabbitMQ
++++++++++++++++++
+
+RabbitMQ health is determined using the command ``rabbitmqctl cluster_status``:
+
+.. code-block:: console
+
+   [stack@controller0 ~]$ docker exec rabbitmq rabbitmqctl cluster_status
+
+   Cluster status of node rabbit@controller0 ...
+   [{nodes,[{disc,['rabbit@controller0','rabbit@controller1',
+                   'rabbit@controller2']}]},
+    {running_nodes,['rabbit@controller1','rabbit@controller2',
+                    'rabbit@controller0']},
+    {cluster_name,<<"rabbit@controller2">>},
+    {partitions,[]},
+    {alarms,[{'rabbit@controller1',[]},
+             {'rabbit@controller2',[]},
+             {'rabbit@controller0',[]}]}]
+
+Checking Keepalived
++++++++++++++++++++
+
+On (for example) three controllers:
+
+.. code-block:: console
+
+   [stack@controller0 ~]$ docker logs keepalived
+
+Two instances should show:
+
+.. code-block:: console
+
+   VRRP_Instance(kolla_internal_vip_51) Entering BACKUP STATE
+
+and the other:
+
+.. code-block:: console
+
+   VRRP_Instance(kolla_internal_vip_51) Entering MASTER STATE
+
+Ansible Control Host
+--------------------
+
+The Ansible control host is not enrolled in bifrost. This node may run services
+such as the seed virtual machine which will need to be gracefully powered down.
+
+Compute
+-------
+
+If you are shutting down a single hypervisor, to avoid down time to tenants it
+is advisable to migrate all of the instances to another machine. See
+:ref:`evacuating-all-instances`.
+
+Ceph
+----
+
+The following guide provides a good overview:
+https://access.redhat.com/documentation/en-us/red_hat_openstack_platform/8/html/director_installation_and_usage/sect-rebooting-ceph
+
+Shutting down the seed VM
+-------------------------
+
+.. code-block:: console
+
+   kayobe# virsh shutdown <Seed hostname>
+
+.. _full-shutdown:
+
+Full shutdown
+-------------
+
+In case a full shutdown of the system is required, we advise to use the
+following order:
+
+* Perform a graceful shutdown of all virtual machine instances
+* Shut down compute nodes
+* Shut down monitoring node (if separate from controllers)
+* Shut down network nodes (if separate from controllers)
+* Shut down controllers
+* Shut down Ceph nodes (if applicable)
+* Shut down seed VM
+* Shut down Ansible control host
+
+Rebooting a node
+----------------
+
+Use ``reboot.yml`` playbook to reboot nodes
+Example: Reboot all compute hosts apart from compute0:
+
+.. code-block:: console
+
+   kayobe# kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/reboot.yml --limit 'compute:!compute0'
```

Comment:
The limit on this command is unnecessarily weird and complicated. I'd change it to something a bit easier to understand
```suggestion
   kayobe# kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/reboot.yml --limit compute4,compute5
```

Reply:
That is valid syntax and I think it's a necessary example.
Users might want to try rebooting all hypervisors except for a few.
If they have large number of hypervisors it would be painful to write every hypervisors to reboot with comma separated list.